### ISSUE TITLE
Ionic & Latest OneSignal-Cordova SDK don't work together.

### ISSUE THREAD

**Issue by selected-pixel-jameson** [OneSignal/OneSignal-Cordova-SDK#388]
**Description:**
When I try to run `ionic cordova run android --device` in Ionic app with the latest OneSignal SDK I get the following error and the app fails to launch. When I downgraded to 2.4.0 it works fine. 

`'Please put onesignal-gradle-plugin first OR update to com.android.tools.build:gradle:3.0.0 or newer!`

**Environment**
1. Running the latest versions
2. 
```
<?xml version='1.0' encoding='utf-8'?>
<widget id="com.keycollectorcomics.keycollector" version="1.5.2" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
    <name>Key Collector</name>
    <description>Key Collector Comics.</description>
    <author email="jameson@selectedpixel.com" href="http://selectedpixel.com">Selected Pixel</author>
    <content src="index.html" />
    <access origin="*" />
    <allow-navigation href="http://ionic.local/*" />
    <allow-intent href="http://*/*" />
    <allow-intent href="https://*/*" />
    <allow-intent href="tel:*" />
    <allow-intent href="sms:*" />
    <allow-intent href="mailto:*" />
    <allow-intent href="geo:*" />
    <allow-navigation href="*" />
    <preference name="android-minSdkVersion" value="19" />
    <preference name="android-targetSdkVersion" value="26" />
    <preference name="webviewbounce" value="false" />
    <preference name="UIWebViewBounce" value="false" />
    <preference name="DisallowOverscroll" value="true" />
    <preference name="BackupWebStorage" value="none" />
    <preference name="SplashMaintainAspectRatio" value="true" />
    <preference name="SplashScreen" value="screen" />
    <preference name="SplashScreenDelay" value="30000" />
    <preference name="AutoHideSplashScreen" value="false" />
    <preference name="SplashShowOnlyFirstTime" value="false" />
    <preference name="FadeSplashScreen" value="true" />
    <preference name="orientation" value="portrait" />
    <platform name="android">
        <allow-intent href="market:*" />
        <icon density="ldpi" src="resources/android/icon/drawable-ldpi-icon.png" />
        <icon density="mdpi" src="resources/android/icon/drawable-mdpi-icon.png" />
        <icon density="hdpi" src="resources/android/icon/drawable-hdpi-icon.png" />
        <icon density="xhdpi" src="resources/android/icon/drawable-xhdpi-icon.png" />
        <icon density="xxhdpi" src="resources/android/icon/drawable-xxhdpi-icon.png" />
        <icon density="xxxhdpi" src="resources/android/icon/drawable-xxxhdpi-icon.png" />
        <splash density="port-ldpi" src="resources/android/splash/drawable-port-ldpi-screen.png" />
        <splash density="port-mdpi" src="resources/android/splash/drawable-port-mdpi-screen.png" />
        <splash density="port-hdpi" src="resources/android/splash/drawable-port-hdpi-screen.png" />
        <splash density="port-xhdpi" src="resources/android/splash/drawable-port-xhdpi-screen.png" />
        <splash density="port-xxhdpi" src="resources/android/splash/drawable-port-xxhdpi-screen.png" />
        <splash density="port-xxxhdpi" src="resources/android/splash/drawable-port-xxxhdpi-screen.png" />
        <splash density="land-ldpi" src="resources/android/splash/drawable-land-ldpi-screen.png" />
        <splash density="land-mdpi" src="resources/android/splash/drawable-land-mdpi-screen.png" />
        <splash density="land-hdpi" src="resources/android/splash/drawable-land-hdpi-screen.png" />
        <splash density="land-xhdpi" src="resources/android/splash/drawable-land-xhdpi-screen.png" />
        <splash density="land-xxhdpi" src="resources/android/splash/drawable-land-xxhdpi-screen.png" />
        <splash density="land-xxxhdpi" src="resources/android/splash/drawable-land-xxxhdpi-screen.png" />
    </platform>
    <platform name="ios">
        <icon height="57" src="resources/ios/icon/icon.png" width="57" />
        <icon height="114" src="resources/ios/icon/icon@2x.png" width="114" />
        <icon height="40" src="resources/ios/icon/icon-40.png" width="40" />
        <icon height="80" src="resources/ios/icon/icon-40@2x.png" width="80" />
        <icon height="120" src="resources/ios/icon/icon-40@3x.png" width="120" />
        <icon height="50" src="resources/ios/icon/icon-50.png" width="50" />
        <icon height="100" src="resources/ios/icon/icon-50@2x.png" width="100" />
        <icon height="60" src="resources/ios/icon/icon-60.png" width="60" />
        <icon height="120" src="resources/ios/icon/icon-60@2x.png" width="120" />
        <icon height="180" src="resources/ios/icon/icon-60@3x.png" width="180" />
        <icon height="72" src="resources/ios/icon/icon-72.png" width="72" />
        <icon height="144" src="resources/ios/icon/icon-72@2x.png" width="144" />
        <icon height="76" src="resources/ios/icon/icon-76.png" width="76" />
        <icon height="152" src="resources/ios/icon/icon-76@2x.png" width="152" />
        <icon height="167" src="resources/ios/icon/icon-83.5@2x.png" width="167" />
        <icon height="29" src="resources/ios/icon/icon-small.png" width="29" />
        <icon height="58" src="resources/ios/icon/icon-small@2x.png" width="58" />
        <icon height="87" src="resources/ios/icon/icon-small@3x.png" width="87" />
        <icon height="1024" src="resources/ios/icon/icon-1024.png" width="1024" />
        <splash height="1136" src="resources/ios/splash/Default-568h@2x~iphone.png" width="640" />
        <splash height="1334" src="resources/ios/splash/Default-667h.png" width="750" />
        <splash height="2208" src="resources/ios/splash/Default-736h.png" width="1242" />
        <splash height="1242" src="resources/ios/splash/Default-Landscape-736h.png" width="2208" />
        <splash height="1536" src="resources/ios/splash/Default-Landscape@2x~ipad.png" width="2048" />
        <splash height="2048" src="resources/ios/splash/Default-Landscape@~ipadpro.png" width="2732" />
        <splash height="768" src="resources/ios/splash/Default-Landscape~ipad.png" width="1024" />
        <splash height="2048" src="resources/ios/splash/Default-Portrait@2x~ipad.png" width="1536" />
        <splash height="2732" src="resources/ios/splash/Default-Portrait@~ipadpro.png" width="2048" />
        <splash height="1024" src="resources/ios/splash/Default-Portrait~ipad.png" width="768" />
        <splash height="960" src="resources/ios/splash/Default@2x~iphone.png" width="640" />
        <splash height="480" src="resources/ios/splash/Default~iphone.png" width="320" />
        <splash height="2732" src="resources/ios/splash/Default@2x~universal~anyany.png" width="2732" />
        <config-file parent="NSCameraUsageDescription" platform="ios" target="*-Info.plist">
            <string>To use the camera on your device to capture images.</string>
        </config-file>
        <config-file parent="NSPhotoLibraryUsageDescription" platform="ios" target="*-Info.plist">
            <string>To use images from your photo library within the application.</string>
        </config-file>
    </platform>
    <branch-config>
        <branch-key value="key_live_gnAO0k2tDVWly5P7SAYCMifjEwa0C87H" />
        <uri-scheme value="keycollectorcomics" />
        <link-domain value="keycollectorcomics.app.link" />
        <link-domain value="keycollectorcomics.test-app.link" />
        <ios-team-release value="CUUL5292QL" />
    </branch-config>
    <plugin name="onesignal-cordova-plugin" spec="2.4.0" />
    <plugin name="cordova-plugin-device" spec="^2.0.2" />
    <plugin name="cordova-plugin-splashscreen" spec="^5.0.2" />
    <plugin name="cordova-plugin-statusbar" spec="^2.4.2" />
    <plugin name="cordova-plugin-whitelist" spec="^1.3.3" />
    <plugin name="cordova-plugin-ionic-keyboard" spec="^2.1.2" />
    <plugin name="cordova-plugin-google-analytics" spec="^1.8.3" />
    <plugin name="cordova-plugin-camera" spec="^3.0.0" />
    <plugin name="cordova-plugin-file" spec="^5.0.0" />
    <plugin name="cordova-plugin-file-transfer" spec="^1.7.1" />
    <plugin name="cordova-plugin-apprate" spec="^1.4.0" />
    <plugin name="branch-cordova-sdk" spec="^2.7.1" />
    <plugin name="cordova-plugin-x-socialsharing" spec="^5.4.1" />
    <plugin name="cordova-plugin-appavailability" spec="^0.4.2" />
    <plugin name="cordova-plugin-inappbrowser" spec="^1.7.2" />
    <plugin name="cordova-plugin-hockeyapp" spec="^5.1.2" />
    <plugin name="cordova-plugin-app-version" spec="^0.1.9" />
    <plugin name="cordova-plugin-stripe" spec="^1.5.3" />
    <plugin name="cordova-custom-config" spec="^5.0.2" />
    <plugin name="cordova-plugin-ionic-webview" spec="^1.2.1" />
    <engine name="android" spec="~6.3.0" />
</widget>
```

**Steps to Reproduce Issue:**
  1. Add the onesignal-cordova-plugin to your project
  2. Build and run android on device.

**Comment by fatsjohonimahnn** [comment]
Please double check that you are using Cordova 8.0.0 or newer.

Also update to the onesignal-cordova-plugin 2.4.1 by removing the plugin and adding it back: 

`ionic cordova plugin remove onesignal-cordova-plugin`
`ionic cordova plugin add onesignal-cordova-plugin`

Please share the config.xml and crash log after checking these.

**Comment by atorgfr** [comment]
@fatsjohonimahnn hi i think it's a problem with the change on build-extras-onesignal.gradle file.

**Comment by jkasten2** [comment]
@selected-pixel-jameson The `onesignal-gradle-plugin` message is only a warning as it might not be able to correctly align dependencies with other plugins. In addition to updating to Cordova CLI 8.0.0 it is recommend to update the Cordova Android Platform to 7.0.0 or newer.
```
cordova platform remove android
cordova platform add android@7.1.0
```

Please include the full runtime error if the above does not fix your issue.

**Comment by princenishchal** [comment]
I am having the same issue with Latest Ionic and OneSignal. In fact a few other projects that we have used Onesignal when recompiled with the latest OneSignal plugin are giving the same errors.

I created an empty "tabs" Ionic project, no additional code just added latest OneSignal Plugin and can reproduce the above error.

Please somebody help fixing this error!

**Comment by princenishchal** [comment]
Hey, I found one way that worked. It is same as what @atorgfr has suggested.

Steps:

Remove the existing onesignal plugin
Add a previous version of the plugin, in my case I added ionic cordova add plugin onesignal-cordova-plugin@2.3.0
Copied the gradle file from the location platforms/android/onesignal-cordova-plugin to a temporary location
After copying this file, I removed this plugin and added the latest Onesignal plugin once again
Replaced the gradle file in location platforms/android/onesignal-cordova-plugin
Compile the project and it worked back fine.
pheww... lots of steps. After so many permutations and combinations this seemed to work.
Special thanks and mentions to @atorgfr . I guess copying from your comments made some mess on the gradle file maybe that's why it did not work.
